envoy/README.md

```diff
@@ -0,0 +1,528 @@
+  rpc GetInitialData(InitialDataRequest) returns (InitialDataResponse);
+
+  // StreamUpdates listens for updates from the remote gRPC server and streams them to the client.
+  rpc StreamUpdates(UpdatesRequest) returns (stream Update);
+}
+```
+
+### 5.2. Remote gRPC Auth Server
+
+The `Remote gRPC Server` is responsible for providing the `Go External Authorization Server` with data on which endpoints are authorized to use the PATH service.
+
+**The implementation of the remote gRPC server is up to the Gateway operator.**
+
+#### 5.2.1. PATH Auth Data Server
+
+The `PATH Auth Data Server (PADS)` is an implementation of the `Remote gRPC Server` that provides `Gateway Endpoint` data to the `Go External Authorization Server` in order to enable authorization for the PATH Gateway.
+
+It is provided to allow an easy default implementation of the remote gRPC server by Gateway operators, or as a starting point for custom implementations.
+
+The Docker image for `PADS` is available at:
+
+```bash
+ghcr.io/buildwithgrove/path-auth-data-server:latest
+```
+
+_This Docker image is loaded by default in the [docker-compose.yml](../docker-compose.yml#L90) file at the root of the PATH repo._
+
+_`PADS` loads data from the `gateway-endpoints.yaml` file specified by the `YAML_FILEPATH` environment variable._
+
+#### 5.2.2. Example Gateway Endpoint YAML File
+
+An example `gateway-endpoints.yaml` file is provided at [envoy/gateway-endpoints.example.yaml](./gateway-endpoints.example.yaml).
+
+```yaml
+endpoints:
+  endpoint_1:
+    endpoint_id: "endpoint_1"
+    auth:
+      require_auth: true
+      authorized_users:
+        "auth0|user_1": {}
+    user_account:
+      account_id: "account_1"
+      plan_type: "PLAN_UNLIMITED"
+
+  endpoint_2:
+    endpoint_id: "endpoint_2"
+    auth:
+      require_auth: false
+    user_account:
+      account_id: "account_2"
+      plan_type: "PLAN_FREE"
+    rate_limiting:
+      throughput_limit: 30
+      capacity_limit: 100000
+      capacity_limit_period: "CAPACITY_LIMIT_PERIOD_MONTHLY"
+```
+
+_In this example, `endpoint_1` is authorized for `user_1` and `endpoint_2` is authorized for all users._
+
+Run `make copy_envoy_gateway_endpoints` to create an example `gateway-endpoints.yaml` file used by the remote gRPC server.
+
+The contents of this file represent the gateway endpoints that are authorized to use the PATH service for a specific gateway operator.
+
+#### 5.2.3. Custom Remote gRPC Server Implementation
+
+If wishing to implement a custom remote gRPC server, the implementation should import the Go `github.com/buildwithgrove/path/envoy/auth_server/proto` package, which is autogenerated from the [`gateway_endpoint.proto`](./auth_server/proto/gateway_endpoint.proto) file.
+
+[The custom implementation must use the methods defined in the `GatewayEndpoints` service:](#51-gateway-endpoints-grpc-service)
+
+- `GetInitialData`
+- `StreamUpdates`
+
+Example simple Go implementation:
+
+_server.go_
+
+```go
```

Comment:
Try to avoid having large code blocks in `.md` files. 

I've found that it gets outdated, stops compiling (no enforcement) so it just gets stale.

Can you move this into a `_example.go` file and have basic compilation enforce consistency?

Reply:
I've removed this as there's not really any need for it anymore.

PADS will be sufficient for most use cases IMO and I've updated the README.md with basic guidelines for a custom implementation (ie. ensure it adheres it to the gateway endpoints proto definition) and a suggestion to fork PADS if wishing to create a custom implementation.